intellij/src/saros/intellij/ui/views/buttons/AbstractSessionToolbarButton.java

```diff
@@ -62,6 +83,21 @@ void setInitialState() {
     }
   }
 
+  @Override
+  public final void dispose() {
+    sarosSessionManager.removeSessionLifecycleListener(sessionLifecycleListener);
+
+    disposeComponents();
+  }
+
+  /**
+   * Method to dispose components of the button.
+   *
+   * <p>This method is called when the project the button belongs to is disposed. It should drop any
+   * internal state that would prevent the object from being garbage collected.
+   */
+  abstract void disposeComponents();
```

Comment:
I do not understand the reason for this method. If you have additional work to do in a subclass you can just override dispose(), either calling super.dispose() at the beginning or at the end of the method.

Reply:
I decided to implement this kind of as a template method to
- ensure that the listener held by the abstract super class is always correctly unregistered (which is why the method `dispose()` in the abstract class is final) and
- hopefully remind the developer to consider whether there is something to dispose when implementing a new button (i.e. avoid error of omission).

This was an active design decision on my part to err on the side of caution (as more buttons will almost definitely have to be added in the future if this is not superseded by the HTML UI beforehand).

But if you think that this is unnecessary/excessively cautious, I can use your suggested approach instead.